tests/functional/models/test_slate_model.py

```diff
@@ -56,39 +91,43 @@ async def test_get_slates_from_slate_configs_without_recs(self):
         assert len(slates[0].recommendations) == 0
 
     async def test_get_slates_from_slate_configs_with_recs(self):
+        self._put_candidates(1)
+
         slates = await SlateModel.get_slates_from_slate_configs([slate_config_model], None)
 
         assert slates[0].id == slate_config_id
         assert len(slates[0].recommendations) == 1
 
+    @patch('app.models.metrics.firefox_new_tab_metrics_factory.FirefoxNewTabMetricsFactory.get',
```

Comment:
It's not great that we have to patch our own stuff in tests. Outside methods, sure, but maybe there's a path forward with dependency injection instead.

I'd be prepared to call that out of scope for this specific ticket, though.

Reply:
Agreed.